### PR TITLE
envydis/gm107: add SUATOM

### PR DESCRIPTION
**Pull request by hakzsam**
It would be good to have a second pair of eyes for this really weird SUATOM instruction on SM50+, because I'm not sure if it's *totallly* correct. :-)

### CODE REVIEW

File: envydis/gm107.c [hunk 1]
```diff
@@ -1595,6 +1595,43 @@ static struct insn tabf0a8_1[] = {
...
+	{ 0x0000001000000000ull, 0x0000007000000000ull, N("s32") },
+	{ 0x0000002000000000ull, 0x0000007000000000ull, N("u64") },
+	{ 0x0000003000000000ull, 0x0000007000000000ull, N("f32.ftz.rn") },
+	{ 0x0000005000000000ull, 0x0000007000000000ull, N("s64") },
+	{ 0x0000006000000000ull, 0x0000007000000000ull, N("sd32") },
+	{ 0x0000007000000000ull, 0x0000007000000000ull, N("sd64") },
+	{ 0, 0, OOPS },
+};
+
+static struct insn tabea80_2[] = {
```
Comment (imirkin) [inline envydis/gm107.c]: Looks like this is incomplete. e.g.

        /*0008*/               @P0 SUATOM.D.1D.ADD.S32.IGN R0, [R0], R0, 0x0;      /* 0xea08000000000000 */
        /*0008*/               @P0 SUATOM.D.1D.ADD.F32.FTZ.RN.IGN R0, [R0], R0, 0x0;      /* 0xea18000000000000 */
        /*0008*/               @P0 SUATOM.D.1D.ADD.S64.IGN R0, [R0], R0, 0x0;      /* 0xea28000000000000 */
        /*0008*/               @P0 SUATOM.D.1D.ADD.SD64.IGN R0, [R0], R0, 0x0;      /* 0xea38000000000000 */

Same values as above, just shifted over by a few bits.

File: envydis/gm107.c [hunk 2]
```diff
@@ -1631,9 +1668,10 @@ static struct insn tabroot[] = {
 	{ 0xeb40000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(      "sured"), T(eb40_0), T(eb40_1), T(eb40_2), T(eb40_3), SUREDMEM, REG_00, T(eb40_4) },
 	{ 0xeb20000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(       "sust"), T(eb40_0), T(eb20_0), T(eb40_1), T(eb20_1), T(eb20_2), T(eb40_3), SUREDMEM, REG_00, T(eb40_4) },
 	{ 0xeb00000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(       "suld"), T(eb40_0),            T(eb40_1), T(eb20_1), T(eb00_0), T(eb40_3), REG_00, SUREDMEM, T(eb40_4) },
-	{ 0xeac0000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(     "suatom") },
-	{ 0xea60000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(     "suatom") },
-	{ 0xea00000000000000ull, 0xff40000000000000ull, OP8B, T(pred), N(     "suatom") },
+	{ 0xeac0000000000000ull, 0xffe0000000000000ull, OP8B, T(pred), N(     "suatom"), T(eb40_0), T(ea80_0), T(ea80_1), T(eb40_1), N("cas"), REG_00, SUREDMEM, REG_20, REG_39 },
```
Comment (imirkin) [inline envydis/gm107.c]: ea80_0 only appears to apply on the D variants

File: envydis/gm107.c [hunk 3]
```diff
@@ -1595,6 +1595,43 @@ static struct insn tabf0a8_1[] = {
...
+	{ 0x0000000010000000ull, 0x0000000010000000ull, N("ba") },
+	{ 0x0000000000000000ull, 0x0000000000000000ull },
+	{ 0, 0, OOPS },
+};
+
+static struct insn tabea80_1[] = {
+	{ 0x0000000000000000ull, 0x0000007000000000ull, N("u32") },
+	{ 0x0000001000000000ull, 0x0000007000000000ull, N("s32") },
+	{ 0x0000002000000000ull, 0x0000007000000000ull, N("u64") },
+	{ 0x0000003000000000ull, 0x0000007000000000ull, N("f32.ftz.rn") },
```
Comment (imirkin) [inline envydis/gm107.c]: Just call this f32. The ftz + rn are implied.

File: envydis/gm107.c (hunk 2 above)
Comment (hakzsam) [inline envydis/gm107.c]: It seems like, but the base opcode is weird. I will figure out :-).

File: envydis/gm107.c (hunk 1 above)
Comment (hakzsam) [inline envydis/gm107.c]: You are right.

File: envydis/gm107.c (hunk 3 above)
Comment (hakzsam) [inline envydis/gm107.c]: Okay.

**Pull request merged**